### PR TITLE
Added migration guide for version 8

### PR DESCRIPTION
**Pull request by surajeaton**
Fixes # BLUI-4359

#### Changes proposed in this Pull Request:
- Added migration guide for v8

#### Screenshots / Screen Recording (if applicable)
-
Screenshot 2024-01-04 at 8 17 10 PM

#### To Test:
- 

#### Any specific feedback you are looking for?
-

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+- [Icon](./docs/Icon.md)
+- [IconSwitch](./docs/IconSwitch.md)
+- [Overline](./docs/Overline.md)
+
+### Removed
+
+- Typography
+- IconWrapper
+
+All the themed components have been removed.
```
Comment (huayunh) [inline MIGRATION.md:46]: So, if I am an adopter, I used tons of `<Typography>` in my apps. Now you tell me all `<Typography>` components have been removed. Ok, I can remove them from my code. Now what? What do I do next?

Comment (surajeaton) [reply MIGRATION.md]: Updated.

File: README.md [hunk 2]
```diff
@@ -73,6 +73,10 @@ to bring up a API documentation website (no interactive components).
 
 See the [documentation](https://brightlayer-ui-components.github.io/react-native) for information on using these components.
 
+### Upgrading to version 8
+
```
Comment (JeffGreiner-eaton) [inline README.md:77]: should we add something here about version 8 in regards to the updated components, material design 3 and our updated theme

File: MIGRATION.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    - This component has been removed from the library can be replaced by [Text](https://callstack.github.io/react-native-paper/docs/components/Text/) component from React Native Paper.
+- IconWrapper
+    - This component has been removed from the library. Icons can be used in Brightlayer UI React Native Components in various ways. Please refer [icons usage guidelines](./docs/Icons) for more info.
+- Themed Components
+    - All the themed components have been removed, as we have updated all the components with the Brightlayer UI's required styles and themes.
+
+## NOTES
+
+<!-- TODO : Add a link to style override guide -->
+There are few React Native Paper compnents which need additional styles. To learn more about style overrides, please follow [the guidelines](#).
```
Comment (JeffGreiner-eaton) [inline MIGRATION.md:53]: typo on components

File: README.md (hunk 2 above)
Comment (JeffGreiner-eaton) [reply README.md:77]: maybe something like this but need some word smithing....

### Upgrading to version 8

Version 8 of [@brightlayer-ui/react-native-components](https://www.npmjs.com/package/@brightlayer-ui/react-native-components) uses React Native Paper v5 with Material Design v3 and is a major update with several changes. Most particularly, the Typography and IconWrapper components have been removed and we added 5 new components. In previous versions we supported Brightlayer-UI React Native Paper Themed Components and they have been removed.

**Comment by huayunh** [comment]
#511

File: MIGRATION.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+First, update your `@brightlayer-ui/react-native-components` dependency to the latest version.
+
+```shell
+npm install --save @brightlayer-ui/react-native-components@^8.0.0
+// or
+yarn upgrade @brightlayer-ui/react-native-components@^8.0.0
+```
+
+You also need to update `@brightlayer-ui/react-native-themes`, `@brightlayer-ui/colors`, `react-native`, and `react-native-paper`.
```
Comment (ektaghag-eaton) [inline MIGRATION.md:13]: The component library doesn't have `@brightlayer-ui/colors` as a dependency.

File: MIGRATION.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+npm install --save @brightlayer-ui/react-native-themes@^7.0.0 @brightlayer-ui/colors@^4.0.0 react-native@0.72.3 react-native-paper@^5.0.0
+
+// or
+
+yarn upgrade @brightlayer-ui/react-native-themes@^7.0.0 @brightlayer-ui/colors@^4.0.0 react-native@0.72.3 react-native-paper@^5.0.0
+```
+
+## Color Updates
+
+The Brightlayer UI color palette underwent major changes to reflect Material Design 3's new color structure. Please follow the [migration guide](https://en.wikipedia.org/wiki/First_solo_flight#Traditions) for more details.
```
Comment (ektaghag-eaton) [inline MIGRATION.md:25]: Is this a correct link?

File: MIGRATION.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    - This component has been removed from the library can be replaced by [Text](https://callstack.github.io/react-native-paper/docs/components/Text/) component from React Native Paper.
+- IconWrapper
+    - Icons no longer requires a wrapper function. Please refer to [icons usage guidelines](./docs/Icons) for more info.
+- Themed Components
+    - All the themed components have been removed, as we have updated all the components with the Brightlayer UI's required styles and themes.
+
+## Notes
+
+<!-- TODO : Add a link to style override guide -->
+A few React Native Paper components require additional style overrides to match Brightlayer UI's design specifications. To learn more about these style overrides, please follow [the guidelines](#).
```
Comment (ektaghag-eaton) [inline MIGRATION.md:55]: Can you add a jira story or issue to keep track of that we need to update the themes link?

**Review by ektaghag-eaton — APPROVED** [review]

File: MIGRATION.md (hunk 5 above)
Comment (huayunh) [reply MIGRATION.md]: ooof, sorry about that

**Pull request merged**